maintenance/wikia/importMessagingWiki.php

```diff
@@ -0,0 +1,195 @@
+			->EQUAL_TO_FIELD( 'rev_id' )
+			->AND_( 'rev_text_id' )
+			->EQUAL_TO_FIELD( 'old_id' );
+
+		# handle --message argument
+		$msg = $this->getOption('message');
+		if ($msg) {
+			# AND page_title LIKE 'Autocreatewiki-welcomebody/%'
+			$sql->AND_('page_title')->LIKE( ucfirst( $msg ) . '/%' );
+		}
+
+		$res = $sql->runLoop( $db, function ( &$res, $row ) use ( &$i, $wgContLang ) {
+			$text = Revision::getRevisionText( $row );
+			$lckey = $wgContLang->lcfirst( $row->page_title );
+			if ( strpos( $lckey, '/' ) ) {
+				$t = explode( '/', $lckey );
+				$key = $t[0];
+				$lang = $t[1];
+			} else {
+				$key = $lckey;
+				$lang = 'en';
+			}
+
+			$res['messages'][$lang] = $res['messages'][$lang] ?? [];
+			$res['messages'][$lang][$key] = $text;
+			$i++;
+		} );
+
+		$this->messages = $res['messages'];
+
+		$this->output( "Found $i messages in messaging DB.\n" );
+		if ( $this->isDryRun ) {
+			$this->output( "Running in dry run mode. Existing i18n files will not be changed.\n" );
+		}
+
+		return !empty( $this->messages );
+	}
+
+	/**
+	 * @return array
+	 */
+	private function getMessagesFiles() {
+		global $wgExtensionMessagesFiles;
+
+		return array_unique(
+			array_merge(
+				GlobalMessagesService::getInstance()->getCoreMessageFiles(),
+				GlobalMessagesService::getInstance()->getExtensionMessageFiles(),
+				$wgExtensionMessagesFiles
+			)
+		);
+	}
+
+	/**
+	 * Process all i18n files of enabled extensions, and update them if messaging wiki has a different version
+	 */
+	public function processExtensions() {
+		foreach ( $this->getMessagesFiles() as $extension => $filePath ) {
+			require $filePath;
+			$changed = 0;
+			/** @var array $messages */
+
+			# is the message defined in this file?
+			# check English so we can handle messages customized on Messaging Wiki that do not exist in i18n files
+			$msg = $this->getOption('message');
+
+			if ($msg) {
+				$msgDefinedInTheCode = isset( $messages['en'][$msg] );
+				if ( $msgDefinedInTheCode === false ) {
+					continue;
+				}
+
+				// we have a single message to process, simply replace i18n files entries here with those from Messaging Wiki
+				// will add missing entries as well (i.e. no entry in the code, translation of theWiki)
+				foreach( $this->messages as $lang => $translations ) {
+					if ( $messages[$lang][$msg] != $translations[$msg] ) {
+						$messages[$lang][$msg] = $translations[$msg];
+						$changed++;
+					}
+				}
+			}
+
+			// iterate over i18n file content and update it using data from Messaging Wiki when neeeded
+			foreach ( $messages as $lang => $translations ) {
+				foreach ( $translations as $key => $text ) {
+					if ( isset( $this->messages[$lang][$key] ) && $this->messages[$lang][$key] !== $messages[$lang][$key] ) {
+						$messages[$lang][$key] = $this->messages[$lang][$key];
+						$changed++;
+					}
+				}
+			}
+
+			if ( $changed === 0 ) {
+				$this->output( "Nothing to update in $extension ($filePath).\n" ); // XXX: DEBUG
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS139)